overlord/snapstate/backend/snapdata.go

```diff
@@ -107,9 +123,72 @@ func copySnapData(oldSnap, newSnap *snap.Info) (err error) {
 	return nil
 }
 
+// trashPath returns the trash path for the given path. This will
+// differ only in the last element.
+func trashPath(path string) string {
+	return path + ".old"
+}
+
+// trash moves path aside, if it exists. If the trash for the path
+// already exists and is not empty it will be removed first.
+func trash(path string) error {
+	trash := trashPath(path)
+	err := os.Rename(path, trash)
+	if err == nil {
+		return nil
+	}
+	// os.Rename says it always returns *os.LinkError. Be wary.
+	e, ok := err.(*os.LinkError)
+	if !ok {
+		return err
+	}
+
+	switch e.Err {
+	case unix.ENOENT:
```

Comment:
yes, `os.IsNotExist(err)` would return `true` for an `*os.LinkError` with `ENOENT`. I'd still have to dig into the `LinkError` for the `ENOTEMPTY` though.